parcels/collection/collectionsoa.py

```diff
@@ -980,6 +980,12 @@ def __init__(self, pcoll, reverse=False, subset=None):
     def __iter__(self):
         return ParticleCollectionIteratorSOA(pcoll=self._pcoll_immutable, reverse=self._reverse, subset=self._subset)
 
+    def __len__(self):
+        return len(self._subset)
+
+    def __getitem__(self, items):
+        return ParticleAccessorSOA(self._pcoll_immutable, self._subset[items])
+
```

Comment:
Please add a docstring to the class member functions and member variables (for variables: a comment) indicating which of them need to be fully defined to comply with particle-particle interaction.